source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.h

```diff
@@ -20,79 +20,106 @@ namespace Extensions {
 namespace StatSinks {
 namespace MetricsService {
 
+using MetricsPtr =
+    std::unique_ptr<Envoy::Protobuf::RepeatedPtrField<io::prometheus::client::MetricFamily>>;
+
 /**
  * Interface for metrics streamer.
  */
-class GrpcMetricsStreamer
-    : public Grpc::AsyncStreamCallbacks<envoy::service::metrics::v3::StreamMetricsResponse> {
+template <class RequestProto, class ResponseProto>
+class GrpcMetricsStreamer : public Grpc::AsyncStreamCallbacks<ResponseProto> {
 public:
+  GrpcMetricsStreamer(Grpc::AsyncClientFactoryPtr&& factory) : client_(factory->create()) {}
```

Comment:
It seems a bit strange to force rvalue reference if this class doesn't actually want ownership? Maybe just pass by ref?

Reply:
I was preserving what was there before, but I agree it is weird. Will change.